### PR TITLE
#588 locked stocktake

### LINKED ISSUE #588: Locked stocktake functionality
Need to add the stocktake locked functionality.

Similar to hold, prevents status changes and deletion

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #588 

- Just adds the toggle button and disables the ok * next status button when on hold
- didnt prevent the deletion - putting on to https://github.com/openmsupply/openmsupply-client/issues/450

### CODE REVIEW

File: packages/inventory/src/Stocktake/DetailView/Footer.tsx [hunk 1]
```diff
@@ -70,6 +71,15 @@ export const Footer: FC<StocktakeDetailFooterProps> = ({ draft, save }) => {
...
             height={64}
           >
+            <ToggleButton
+              disabled={!isStocktakeEditable(draft)}
+              value={!!draft.onHold}
+              selected={!!draft.onHold}
+              onClick={() => {
+                draft.updateOnHold();
+              }}
+              label={t('label.hold')}
```
Comment (mark-prins) [inline packages/inventory/src/Stocktake/DetailView/Footer.tsx:81]: In mSupply the label is `Locked` though I prefer having the name consistent and keeping with `Hold`

**Review by mark-prins — APPROVED** [review]
👍  very nice!

**Comment by mark-prins** [comment]
another thought which I forgot to mention - the `Locked` status should also prevent adding lines. In fact, nothing can be changed; when all the various PRs are merged, we should set the date and inputs to disabled too. maybe disable the save button too?

**Comment by joshxg** [comment]
Oh! Hmm. Yes will get it done after these are merged. Possibly `locked` is more suitable over `hold` if the functionality is different 🤔

**Pull request merged**